src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -275,6 +275,12 @@ public function initialize(array $query = array(), array $request = array(), arr
         $this->basePath = null;
         $this->method = null;
         $this->format = null;
+
+        // if the request doesn't contain a unique ID, generate a random one
+        if (!$this->headers->has('X_REQUEST_ID')) {
+            $uuid = sprintf('%04x%04x-%04x-%04x-%04x-%04x%04x%04x', mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0x0fff) | 0x4000,  mt_rand(0, 0x3fff) | 0x8000, mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0xffff));
+            $this->headers->set('X_REQUEST_ID', $uuid);
+        }
```

Comment:
Would it be an idea to use an existing package for this/create one or is it overkill? uuid is fairly common but neither symfony or doctrine have an out-of-the-box version available

*correct me if I'm wrong*